cmd/dagger/listen.go

```diff
@@ -26,7 +26,7 @@ var listenCmd = &cobra.Command{
 func Listen(cmd *cobra.Command, args []string) {
 	ctx := context.Background()
 	if err := withEngine(ctx, "", func(ctx context.Context, r *router.Router) error {
-		fmt.Fprintf(os.Stderr, "==> server listening on %s\n", listenAddress)
+		fmt.Fprintf(os.Stderr, "==> server listening on http://%s/query\n", listenAddress)
```

Comment:
`listenAddress` defaults to `:8080`, so this would be confusing in that case. Maybe we should just change that to be `127.0.0.1:8080` instead?

Reply:
Shall we do `0.0.0.0` instead of `127.0.0.1`? Since that's the same behavior as `ListenAndServe(":8080")`

Reply:
I don't think clients will be able to connect to 0.0.0.0, that only works when setting up the listener.

The other option would be to not use `http.ListenAndServe` and instead explicitly call `net.Listen`, get the final address (via `l.Addr().(*net.TCPAddr)`) and then print that (and pass it to `http.Serve`))

Reply:
> I don't think clients will be able to connect to 0.0.0.0, that only works when setting up the listener.

In linux `0.0.0.0` currently returns `127.0.0.1`. Mostly the reason why I suggested that, but I agree that it might not be portable. I'll fallback to the `Serve` strategy and use `l.Addr()` instead since it looks more solid

Reply:
Fun fact is that if you use `0.0.0.0:8080` or `:8080` as `listenAddress` `l.Addr()` returns `http://[::]:8080/query` :P

Reply:
> In linux 0.0.0.0 currently resolves to 127.0.0.1. Mostly the reason why I suggested that, but I agree that it might not be portable. I'll fallback to the Serve strategy and use l.Addr() instead since it looks more solid

Oh interesting, I didn't know Linux supported that for connect. Weird cause in the context of a network gateway and listen it means something completely different. Agree it's best to avoid due to not being portable but good to know either way.

> Fun fact is that if you use 0.0.0.0:8080 or :8080 as listenAddress l.Addr() returns http://[::]:8080/query. So for this case calling ListenAndServe or just Serve it's pretty much the same. I'll just change listenAddress to be 127.0.0.1:8080 by default and that's it.

Yeah I ran into that recently with the other listeners: https://github.com/dagger/dagger/pull/4159

Agree it's easiest to just use 127.0.0.1 for now